v2/client.go

```diff
@@ -0,0 +1,206 @@
+package jsonrpc
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+var (
+	// DefaultClient is the client that fulfills the the Call method
+	DefaultClient = NewClient()
+)
+
+// Client is a JSONRPC client that faciliates the calling of methods on a
+// JSONRPC server.
+type Client struct {
+	HTTPClient *http.Client
+}
+
+// NewClient creates a new client that makes use of the http.DefaultClient as
+// it's underlying HTTPClient
+func NewClient() *Client {
+	client := &Client{
+		HTTPClient: http.DefaultClient,
+	}
+	return client
+}
+
+func (client *Client) do(req *http.Request, result interface{}) error {
+
+	rawresp, err := client.HTTPClient.Do(req)
+
+	if err != nil {
+		return err
+	}
+
+	body, err := ioutil.ReadAll(rawresp.Body)
+
+	if err != nil {
+		return err
+	}
+
+	var resp clientResponse
+
+	err = json.Unmarshal(body, &resp)
+
+	if err != nil {
+		return err
+	}
+
+	if resp.Error != nil {
+		return resp.Error
+	}
+
+	err = json.Unmarshal(resp.Result, result)
```

Comment:
specifically `resp.Result` is a `json.RawMessage` which means that the json decoder has parsed it's tokens, but not done any reflection (the slow bit) to convert the raw json into a go type. we can then unmarshal the RawMessage later on once we have a clearer picture of what we want to unmarshal it into. `json.RawMessage` has a type of `[]byte` if that makes it clearer.